mlflow/tracing/client.py

```diff
@@ -342,6 +369,28 @@ def download_trace_extra_fields(trace_info: TraceInfo) -> Trace | None:
 
         return PagedList(traces, next_token)
 
+    def _split_trace_infos_by_trace_data_location(
+        self, trace_infos: list[TraceInfo]
+    ) -> tuple[list[TraceInfo], list[TraceInfo]]:
+        """
+        Split the trace infos where the trace data is stored.
```

Comment:
Can we make the API name and docstring even more specific? We're not splitting by different types of locations, right? We're splitting by (1) traces whose spans are stored in a database or catalog table and (2) traces whose spans are stored in blob storage.